### PR TITLE
Change result structure and fix types

### PR DESCRIPTION
**Pull request by Xstoudi**

### CODE REVIEW

File: src/__tests__/index.ts [hunk 1]
```diff
@@ -17,28 +17,39 @@ test('can pass one check', async () => {
...
-    },
+    ],
   });
 });
+
+test('typescript autocomplete correctly', async () => {
+  const result = await isMyFileReady(file, [endsWithStr(end), sameSize(size)]);
+  const check = result.checks[1];
+  if (check.name === 'endsWithStr') {
+    expect(check.endsWith).toBe(end);
```
Comment (targos) [inline src/__tests__/index.ts:54]: can you check the right array element and add `expect.assertions()` to make sure it's called?

Comment (Xstoudi) [reply src/__tests__/index.ts:54]: Done

File: src/checks/endsWithBytes.ts [hunk 2]
```diff
@@ -2,16 +2,14 @@ import { open, stat } from 'fs/promises';
 
 import { CheckFunction } from '../types';
 
-export function endsWithByteArray(
-  endByteArray: ArrayLike<number>,
-): CheckFunction {
-  if (!Array.isArray(endByteArray)) {
-    throw new TypeError('endByteArray should be an array');
+export function endsWithBytes(endBytes: ArrayLike<number>): CheckFunction {
+  if (!Array.isArray(endBytes)) {
```
Comment (targos) [inline src/checks/endsWithBytes.ts]: This is going to fail if `endBytes` is an Uint8Array. can you assert that the length is at least 1 ? Actually we should also check that in the endsWithStr function

File: src/checks/__tests__/endsWithBytes.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ test('returns the expected result if file does end with the expected string', as
   });
 });
 
-test('throws error if input is not an array of byte', async () => {
+test('throws error if input is not an array of bytes', async () => {
   const t = async () => {
     await endsWithBytes((invalidEnd as unknown) as ArrayLike<number>)(file);
```
Comment (targos) [inline src/checks/__tests__/endsWithBytes.ts]: you can use `@ts-expect-error` for tests that pass wrong types

**Pull request merged**